src/trade.c

```diff
@@ -0,0 +1,4501 @@
+                "\tmovs r1, 0x82\n"
+                "\tlsls r1, 5\n"
+                "\tmovs r0, 0\n"
+                "\tbl SetGpuReg\n"
+                "\tmovs r0, 0x2\n"
+                "\tbl sub_804D694\n"
+                "\tldr r1, _0804CE98 @ =gMain\n"
+                "\tmovs r7, 0x87\n"
+                "\tlsls r7, 3\n"
+                "\tadds r1, r7\n"
+                "\tb _0804CEC2\n"
+                "\t.align 2, 0\n"
+                "_0804CE98: .4byte gMain\n"
+                "_0804CE9C:\n"
+                "\tmovs r0, 0\n"
+                "\tbl sub_804F890\n"
+                "\tldr r1, _0804CEAC @ =gMain\n"
+                "\tmovs r0, 0x87\n"
+                "\tlsls r0, 3\n"
+                "\tadds r1, r0\n"
+                "\tb _0804CEC2\n"
+                "\t.align 2, 0\n"
+                "_0804CEAC: .4byte gMain\n"
+                "_0804CEB0:\n"
+                "\tmovs r0, 0x1\n"
+                "\tbl sub_804F890\n"
+                "\tbl sub_804F964\n"
+                "_0804CEBA:\n"
+                "\tldr r1, _0804CECC @ =gMain\n"
+                "\tmovs r2, 0x87\n"
+                "\tlsls r2, 3\n"
+                "\tadds r1, r2\n"
+                "_0804CEC2:\n"
+                "\tldrb r0, [r1]\n"
+                "\tadds r0, 0x1\n"
+                "\tstrb r0, [r1]\n"
+                "\tb _0804CEE6\n"
+                "\t.align 2, 0\n"
+                "_0804CECC: .4byte gMain\n"
+                "_0804CED0:\n"
+                "\tldr r0, _0804CF08 @ =gPaletteFade\n"
+                "\tldrb r1, [r0, 0x7]\n"
+                "\tmovs r0, 0x80\n"
+                "\tands r0, r1\n"
+                "\tcmp r0, 0\n"
+                "\tbne _0804CEE6\n"
+                "\tldr r0, _0804CF0C @ =sub_804DFF0\n"
+                "\tstr r0, [r3]\n"
+                "\tldr r0, _0804CF10 @ =sub_804D638\n"
+                "\tbl SetMainCallback2\n"
+                "_0804CEE6:\n"
+                "\tbl RunTextPrinters\n"
+                "\tbl RunTasks\n"
+                "\tbl AnimateSprites\n"
+                "\tbl BuildOamBuffer\n"
+                "\tbl UpdatePaletteFade\n"
+                "\tadd sp, 0x28\n"
+                "\tpop {r3}\n"
+                "\tmov r8, r3\n"
+                "\tpop {r4-r7}\n"
+                "\tpop {r0}\n"
+                "\tbx r0\n"
+                "\t.align 2, 0\n"
+                "_0804CF08: .4byte gPaletteFade\n"
+                "_0804CF0C: .4byte sub_804DFF0\n"
+                "_0804CF10: .4byte sub_804D638");
+}
+#endif //NONMATCHING
+
+#ifdef NONMATCHING
+// Nonmatching behaviour is the same as the function above
+void sub_804CF14(void)
+{
+    int i;
+    struct SpriteTemplate temp;
+    u8 id;
+    s32 width;
+    u32 xPos;
+
+    switch (gMain.state)
+    {
+    case 0:
+        sub_804C600();
+        gMain.state++;
+        break;
+    case 1:
+        gMain.state++;
+        sTradeMenuResourcesPtr->unk_A8 = 0;
+        break;
+    case 2:
+        gMain.stat
```

Comment:
For function call in multiple lines please be consistent with the location of first argument. e.g. line 857
And please check other instances in this file

Reply:
Leaving this until the functions in question can be matched